File: Ruby/Civitas/lib/sorpresa_ir_carcel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class SorpresaIrCarcel < Sorpresa
    def initialize(tablero)
      super()
      @texto="Vas a la cárcel. \n"
      @tablero=tablero
    end
  
    # Método sobrescrito
    def aplicar_a_jugador(actual, todos)
      if(jugador_correcto(actual, todos))
        todos[actual].encarcelar(@tablero.num_casilla_carcel)
      end
    end
    
    
    def to_s
       "\n\t ¡SORPRESA! \n
       \n Has recibido: 
       \n #{@texto} \n"
    end
  end
end

File: Java/P4/Civitas/src/civitas/SorpresaIrCarcel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author arturo
 */
public class SorpresaIrCarcel extends Sorpresa{
    private Tablero tablero;
    
    SorpresaIrCarcel(Tablero tablero){
        super("Vas a la carcel\n");
        this.tablero=tablero;
    }
    
    @Override 
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(this.jugadorCorrecto(actual, todos)){
            todos.get(actual).encarcelar(tablero.getCarcel()); 
        }
    }
    
    @Override
    public String toString(){
        return "Sorpresa: "+super.getTexto()+"\n";
    }
}
